enterprise/server/raft/replica/replica.go

```diff
@@ -1695,6 +1513,52 @@ func (sm *Replica) ApplySnapshotFromReader(r io.Reader, db ReplicaWriter) error
 	return nil
 }
 
+type Record struct {
+	PB    proto.Message
+	Error error
+}
+
+func (sm *Replica) ParseSnapshot(ctx context.Context, r io.Reader) <-chan Record {
+	ch := make(chan Record, 10)
+
+	readBuf := bufio.NewReader(r)
+	go func() {
+		defer close(ch)
+		for {
+			r, count, err := readDataFromReader(readBuf)
+			if err != nil {
+				if err == io.EOF {
+					break
+				}
+				ch <- Record{Error: err}
```

Comment:
looks like you are missing `continue` here and for the other error cases in this function

Reply:
Added break -- any error in this function stops the show